src/library/templates/library/edit_customization.html

```diff
@@ -103,5 +126,146 @@ <h2>Edit {{ customization.title }} {% if is_new == 'true' %}(New){% endif %}</h2
        $input.val($input.val() + text);
        $input.focus();
     });
+
+    $('#add-word-button').click(function (e) {
+        addVocabularyWord(e);
+    });
+
+    $('#id_word_to_add').keypress(function (e) {
+        var keycode = (e.keyCode ? event.keyCode : event.which);
+        if (keycode == '13') {
+            addVocabularyWord(e);
+        }
+    });
+
+    $('.delete-word-button').click(function (e) {
+        deleteVocabularyWord(e);
+    });
+
+    function deleteVocabularyWord (e) {
+        e.preventDefault();
+        var parentDiv = $(e.currentTarget).parent();
+        var wordToDelete= $(e.currentTarget).attr('id').replace(/^delete_/,'').replace('_', ' ');
+        // Remove wordToDelete from the new words list, if present.
+        var newResult = isNewWord(wordToDelete);
+        if (newResult.isMember) {
+            var modNewWords = $.grep(newResult.wordArray, function (aWord) {
+                return aWord !== wordToDelete;
+            });
+            $('#id_new_vocabulary_words').val(modNewWords.join('|').trim());
+        }
+        else {
+            // Note:  leave the current words list intact as it records the
+            // actual current words -- a current word can appear in both the
+            // current words list and the deleted words list, but a new word is
+            // either in the new words list or not, and never in the deleted
+            // words list.
+            // Add wordToDelete to deleted words, but only once.
+            var deleteResult = isDeletedWord(wordToDelete);
+            if (!deleteResult.isMember) {
+                deleteResult.wordArray.push(wordToDelete);
+                $('#id_delete_vocabulary_words').val(deleteResult.wordArray.join('|').trim());
+            }
+        }
+        // In all cases, removed the <div> containint the wordToDelete from the
+        // document.
+        $(parentDiv).remove();
+        e.stopPropagation();
+    };
+
+    function addVocabularyWord(e) {
+        e.preventDefault();
+        e.stopPropagation();
+        var inputField = $('#id_word_to_add');
+        var newWord = inputField.val().trim();
+        if (newWord.length) {
+            var condition = addCondition(newWord);
+            if (condition === 'addNew' || condition === 'addAndRemoveFromDeleted') {
+                var delete_button_id = 'delete_' + newWord.replace(' ', '_');
+                // Add the new <div> for the "new" vocabulary word.
+                $('.vocabulary-word-list').append(`
+                    <div class="col-lg-4 vocabulary-word new-vocabulary-word">
+                        ${newWord}  <button id="${delete_button_id}" class="link-btn">x</button>
+                    </div>
+                `);
+                // Hook up the delete listener for the new word's delete button.
```

Comment:
Using a [delegated handler](https://learn.jquery.com/events/event-delegation/) would avoid the need for this.

Reply:
A previous version had the handler higher up and relied on bubbling, but, whenever a bug occurred the click would bubble such that the form would submit and navigate away from the edit customization page -- there was little information left around to indicate what went wrong.  To counter that, the handler stopped propagation right at the delete button, and that entailed having a handler for each delete button.

But now that things are more stable, yes, a single handler is better, and put it higher up in chain.